src/broker/broker_log_converter.c

```diff
@@ -278,135 +278,188 @@ log_bind_value (char *str, int bind_len, int lineno, FILE * outfp)
 	}
     }
 
-  if (strcmp (p, "NULL") == 0)
+  type = CCI_U_TYPE_LAST + 1;
+  switch (*p)
     {
-      type = CCI_U_TYPE_NULL;
-    }
-  else if (strcmp (p, "CHAR") == 0)
-    {
-      type = CCI_U_TYPE_CHAR;
-    }
-  else if (strcmp (p, "VARCHAR") == 0)
-    {
-      type = CCI_U_TYPE_STRING;
-    }
-  else if (strcmp (p, "NCHAR") == 0)
-    {
-      type = CCI_U_TYPE_NCHAR;
-    }
-  else if (strcmp (p, "VARNCHAR") == 0)
-    {
-      type = CCI_U_TYPE_VARNCHAR;
-    }
-  else if (strcmp (p, "BIT") == 0)
-    {
-      type = CCI_U_TYPE_BIT;
-    }
-  else if (strcmp (p, "VARBIT") == 0)
-    {
-      type = CCI_U_TYPE_VARBIT;
-    }
-  else if (strcmp (p, "NUMERIC") == 0)
-    {
-      type = CCI_U_TYPE_NUMERIC;
-    }
-  else if (strcmp (p, "UBIGINT") == 0)
-    {
-      type = CCI_U_TYPE_UBIGINT;
-    }
-  else if (strcmp (p, "BIGINT") == 0)
-    {
-      type = CCI_U_TYPE_BIGINT;
-    }
-  else if (strcmp (p, "UINT") == 0)
-    {
-      type = CCI_U_TYPE_UINT;
-    }
-  else if (strcmp (p, "INT") == 0)
-    {
-      type = CCI_U_TYPE_INT;
-    }
-  else if (strcmp (p, "USHORT") == 0)
-    {
-      type = CCI_U_TYPE_USHORT;
-    }
-  else if (strcmp (p, "SHORT") == 0)
-    {
-      type = CCI_U_TYPE_SHORT;
-    }
-  else if (strcmp (p, "MONETARY") == 0)
-    {
-      type = CCI_U_TYPE_MONETARY;
-    }
-  else if (strcmp (p, "FLOAT") == 0)
-    {
-      type = CCI_U_TYPE_FLOAT;
-    }
-  else if (strcmp (p, "DOUBLE") == 0)
-    {
-      type = CCI_U_TYPE_DOUBLE;
-    }
-  else if (strcmp (p, "DATE") == 0)
-    {
-      type = CCI_U_TYPE_DATE;
-    }
-  else if (strcmp (p, "TIME") == 0)
-    {
-      type = CCI_U_TYPE_TIME;
-    }
-  else if (strcmp (p, "TIMESTAMP") == 0)
-    {
-      type = CCI_U_TYPE_TIMESTAMP;
-    }
-  else if (strcmp (p, "DATETIME") == 0)
-    {
-      type = CCI_U_TYPE_DATETIME;
-    }
-  else if (strcmp (p, "TIMESTAMPTZ") == 0)
-    {
-      type = CCI_U_TYPE_TIMESTAMPTZ;
-    }
-  else if (strcmp (p, "DATETIMETZ") == 0)
-    {
-      type = CCI_U_TYPE_DATETIMETZ;
-    }
-  else if (strcmp (p, "OBJECT") == 0)
-    {
-      type = CCI_U_TYPE_OBJECT;
-    }
-  else if (strcmp (p, "BLOB") == 0)
-    {
-      type = CCI_U_TYPE_NULL;
-      fprintf (stderr, "%s\nBLOB type is not implemented. Replaced with NULL\n", value_p);
-      value_p = (char *) "";
-      bind_len = 0;
-      /* type = CCI_U_TYPE_BLOB; */
-    }
-  else if (strcmp (p, "CLOB") == 0)
-    {
-      type = CCI_U_TYPE_NULL;
-      fprintf (stderr, "%s\nCLOB type is not implemented. Replaced with NULL\n", value_p);
-      value_p = (char *) "";
-      bind_len = 0;
-      /* type = CCI_U_TYPE_CLOB; */
-    }
-  else if (strcmp (p, "ENUM") == 0)
-    {
-      type = CCI_U_TYPE_ENUM;
-    }
-  else if (strcmp (p, "JSON") == 0)
-    {
-      type = CCI_U_TYPE_JSON;
+    case 'B':
+      if (memcmp (p, "BIGINT", 7) == 0)
```

Comment:
```suggestion
      if (memcmp (p, "BIGINT", 6) == 0)
```

memcmp 함수의 세 번째 인자 값이 broker_log_converter.c, broker_log_replay.c 와 broker_log_util.c 간에 다르게 사용되고 있는 것 같습니다.

converter와 replay 쪽에서는 **문자 길이 + 1 (널 종료 문자 포함?)** 로 비교하고 있고,

util 쪽에서는 **문자 길이만큼** 만 비교하고 있습니다.

해당 차이가 의도된 처리인지 한 번 확인 부탁드립니다.

Reply:
strncmp()는 지정된 길이까지를 비교합니다만 strcmp()는  문자열 끝의 널문자까지 비교해 줘야 합니다.
strncmp ("CHAR", "CHARS", 4)는 비교 결과 동일하다고 판단합니다.  memcmp ("CHAR", "CHARS", 4)와 같습니다.
strcmp ( "CHAR", "CHARS")는 비교 결과가 다르다고 판단합니다.   memcmp ("CHAR", "CHARS", 5)와 같습니다.
널 문자까지 비교해 보아야 하기 때문에 `+ 1` 을 해 준 것입니다.

Reply:
네 이해했습니다. 리뷰 감사합니다.